### PR TITLE
[ecowatt] Initial contribution

### LINKED ISSUE #13351: Implement Ecowatt Binding
I plan to create a binding for Ecowatt : https://www.monecowatt.fr
Of course, this would be something of interest only for the French users.
On the Website, they announce a simplified API to be released during the month of September. So I will wait for this new version.
The only small "challenge" will be the required OAuth2 authentication.
@clinique : you could be interested ? Maybe you are already working on that ?

### PR DESCRIPTION
**Pull request by lolodomo**
This binding uses the Ecowatt API to expose clear signals to adopt the right gestures and to ensure a good supply of electricity for all in France.

Close #13351

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
Here is a jar (rename the extension) for testing:
[org.openhab.binding.ecowatt-3.4.0-SNAPSHOT.zip](https://github.com/openhab/openhab-addons/files/9593472/org.openhab.binding.ecowatt-3.4.0-SNAPSHOT.zip)

File: bundles/org.openhab.binding.ecowatt/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+			</parameter>
+			<parameter name="idSecret" type="text" required="true">
+				<context>password</context>
+				<label>ID Secret</label>
+				<description>ID secret provided with the application you created in the RTE portal.</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- Sample Channel Type -->
```
Comment (fwolter) [inline bundles/org.openhab.binding.ecowatt/src/main/resources/OH-INF/thing/thing-types.xml:42]:
```suggestion
```

**Review by fwolter — APPROVED** [review]
LGTM. Let's merge this fast to generate a small contribution in fixing the current energy crises.

**Comment by lolodomo** [comment]
> LGTM. Let's merge this fast to generate a small contribution in fixing the current energy crises.

Thank you Fabian.

**Pull request merged**